CommonApplication.cmake

```diff
@@ -38,62 +39,66 @@ function(common_application Name)
   cmake_parse_arguments(THIS "${_opts}" "${_singleArgs}" "${_multiArgs}"
     ${ARGN})
 
+  if(THIS_WIN32 AND MSVC)
+    set(_options WIN32)
+  elseif(THIS_GUI AND APPLE)
+    set(_options MACOSX_BUNDLE)
+  endif()
+
   string(TOUPPER ${Name} NAME)
   string(TOLOWER ${Name} name)
-  set(SOURCES ${${NAME}_SOURCES})
-  set(HEADERS ${${NAME}_HEADERS})
-  set(LINK_LIBRARIES ${${NAME}_LINK_LIBRARIES})
-  set(_ICON ${${NAME}_ICON}) # also used to configure Info.plist
+  set(_sources ${${NAME}_SOURCES})
+  set(_headers ${${NAME}_HEADERS})
+  set(_libraries ${${NAME}_LINK_LIBRARIES})
 
   if(${NAME}_SHADERS)
     stringify_shaders(${${NAME}_SHADERS})
-    list(APPEND SOURCES ${SHADER_SOURCES})
+    list(APPEND _sources ${SHADER_SOURCES})
   endif()
 
-  set(OPTIONS)
   if(THIS_GUI)
     if(APPLE)
-      set(OPTIONS MACOSX_BUNDLE)
+      set(_icon_file_candidate ${name}.icns)
+    elseif(NOT MSVC)
+      set(_icon_file_candidate ${name}.png)
+      set(_desktop_file_candidate ${name}.desktop)
+      if(DEFINED ${NAME}_DESKTOP)
+        set(_desktop ${${NAME}_DESKTOP})
+      elseif(EXISTS ${CMAKE_CURRENT_LIST_DIR}/${_desktop_file_candidate})
+        set(_desktop ${_desktop_file_candidate})
+      endif()
     endif()
-  endif()
 
-  if(THIS_WIN32)
-    if(MSVC)
-      set(OPTIONS WIN32)
+    # _ICON is also used to configure Info.plist on OSX
+    # EXISTS only works with absolute file path
+    if(DEFINED ${NAME}_ICON)
+      set(_ICON ${${NAME}_ICON})
+    elseif(EXISTS ${CMAKE_CURRENT_LIST_DIR}/${_icon_file_candidate})
+      set(_ICON ${_icon_file_candidate})
```

Comment:
yes, I used lower case and added set(_ICON ${_icon})
before configuring the Info.plist file, it is much clearer that way